duckpgq/src/duckpgq/functions/tablefunctions/pgq_scan.cpp

```diff
@@ -38,6 +39,47 @@ static void ScanCSREFunction(ClientContext &context, TableFunctionInput &data_p,
   FlatVector::SetData(output.data[0], (data_ptr_t)csr->e.data());
 }
 
+static void ScanCSRPtrFunction(ClientContext &context, TableFunctionInput &data_p,
+                             DataChunk &output) {
+  bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+  if (gstate) {
+    output.SetCardinality(0);
+    return;
+  }
+
+  gstate = true;
+
+  auto duckpgq_state_entry = context.registered_state.find("duckpgq");
+  if (duckpgq_state_entry == context.registered_state.end()) {
+    //! Wondering how you can get here if the extension wasn't loaded, but
+    //! leaving this check in anyways
+    throw MissingExtensionException(
+        "The DuckPGQ extension has not been loaded");
+  }
+  auto duckpgq_state =
+      reinterpret_cast<DuckPGQState *>(duckpgq_state_entry->second.get());
+  auto csr_id = data_p.bind_data->Cast<CSRScanEData>().csr_id;
+  CSR *csr = duckpgq_state->GetCSR(csr_id);
+  output.SetCardinality(5);
+  output.data[0].SetVectorType(VectorType::FLAT_VECTOR);
+  //output.data[0].SetVectorType(VectorType::CONSTANT_VECTOR);
+  auto result_data = FlatVector::GetData<uint64_t>(output.data[0]);
+  result_data[0] = (uint64_t)(csr->v);
```

Comment:
Could you add a couple of comments on what is happening here? I'd understand now but not in a couple of months